### ISSUE TITLE
8.8.3 Causes sharp module issue

### ISSUE THREAD

**Issue by ezos86** [infinitered/ignite#2483]
### Describe the bug

When running the create app command. The sharp module has an issue. Something went wrong installing the "sharp" module.

`npx ignite-cli@latest new PizzaApp --yes`

<img width="834" alt="image" src="https://github.com/infinitered/ignite/assets/1334598/b2fe7f2c-0e3d-4956-97ab-74fce0967c7f">

### Ignite version

8.8.3

### Additional info

I believe the sharp package needs to be updated in the new module for 8.8.3

**Comment by frankcalise** [comment]
Hi @ezos86 - thanks for the report. Please try the latest version of Ignite which is 8.84 and let us know if the same thing occurs. Could you provide the output of `npx ignite-cli doctor` as well as run your new command again with added `--debug` flag to help determine what is going on? For example: `npx ignite-cli@latest new PizzaApp --yes --debug`.

Also, please consult this issue https://github.com/infinitered/ignite/issues/2468

**Comment by joshuayoes** [comment]
@ezo86 can you check your system `sharp` install? Also, are you on an M1? There seems to be some issues with Sharp on M1s https://github.com/lovell/sharp/issues/2794

**Comment by frankcalise** [comment]
Closing this due to no response, please feel free to re-open should you revisit the issue. But please check your global `sharp` module as the project does not have a specific installed dependency on this where we could update it.

**Comment by azinedine** [comment]
i have the same issue here 


user@users-MBP Mushaf % npx ignite-cli@latest new mushafApp
/Users/user/.npm/_npx/bec2839915ba8fc3/node_modules/gluegun/build/index.js:15
    throw up;
    ^

Error:
Something went wrong installing the "sharp" module

Cannot find module '../build/Release/sharp-darwin-arm64v8.node'
Require stack:
- /Users/user/.npm/_npx/bec2839915ba8fc3/node_modules/sharp/lib/sharp.js
- /Users/user/.npm/_npx/bec2839915ba8fc3/node_modules/sharp/lib/constructor.js
- /Users/user/.npm/_npx/bec2839915ba8fc3/node_modules/sharp/lib/index.js
- /Users/user/.npm/_npx/bec2839915ba8fc3/node_modules/ignite-cli/build/tools/generators.js
- /Users/user/.npm/_npx/bec2839915ba8fc3/node_modules/ignite-cli/build/commands/help.js
- /Users/user/.npm/_npx/bec2839915ba8fc3/node_modules/ignite-cli/build/cli.js
- /Users/user/.npm/_npx/bec2839915ba8fc3/node_modules/ignite-cli/bin/ignite

Possible solutions:
- Install with verbose logging and look for errors: "npm install --ignore-scripts=false --foreground-scripts --verbose sharp"
- Install for the current darwin-arm64v8 runtime: "npm install --platform=darwin --arch=arm64v8 sharp"
- Consult the installation documentation: https://sharp.pixelplumbing.com/install
    at Object.<anonymous> (/Users/user/.npm/_npx/bec2839915ba8fc3/node_modules/sharp/lib/sharp.js:34:9)
    at Module._compile (node:internal/modules/cjs/loader:1256:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1310:10)
    at Module.load (node:internal/modules/cjs/loader:1119:32)
    at Module._load (node:internal/modules/cjs/loader:960:12)
    at Module.require (node:internal/modules/cjs/loader:1143:19)
    at require (node:internal/modules/cjs/helpers:110:18)
    at Object.<anonymous> (/Users/user/.npm/_npx/bec2839915ba8fc3/node_modules/sharp/lib/constructor.js:8:1)
    at Module._compile (node:internal/modules/cjs/loader:1256:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1310:10)

Node.js v18.17.0

**Comment by frankcalise** [comment]
@azinedine can you provide your `npx ignite-cli doctor` output?

**Comment by azinedine** [comment]
I changed my Node.js version from v18.17.0 to v20.2.0, and it's working.

**Comment by ceres-quangd** [comment]
I'm still getting this error. 
Node v20.8.0

**Comment by ipublicweb** [comment]
As latest app requires nodejs >= 18, Sharp error could appear when node x86 is installed, instead of x64 version

**Comment by markrickert** [comment]
I ran into this today just trying to run `yarn` on a cloned version of the main repo (not trying to build an ignited app).

I was able to get past it by installing `sharp`and `node-gyp` directly and globally:

```
npm install sharp node-gyp -g
```

**Comment by stek93** [comment]
If anyone is still encountering issues like this, I wanted to share what worked for me.

I’m on an Apple Silicon Mac (M1 Pro) and was using Node.js v22 when I ran into the `sharp` module error while trying to create a new Ignite project. The error complained about a missing binary (`sharp-darwin-arm64v8.node`).

After some digging, I found that Ignite CLI uses a temporary npx cache to install dependencies, and it wasn’t pulling the right binary for my architecture. Here’s what fixed it:

1. **Locate the `sharp` module in the npx cache** – I checked where it was installed:

    ```
    ls -l /Users/<username>/.npm/_npx/*/node_modules/sharp/build/Release
    ```
    This let me confirm that the wrong binary (`sharp-darwin-x64.node`) was present instead of the one I needed (`sharp-darwin-arm64v8.node`).
    
 2. **Navigate to the `sharp` directory** – I went into the cached folder:
 
    ```
     cd /Users/<username>/.npm/_npx/*/node_modules/sharp
     ```
   
 3. **Force a rebuild** – I ran this command to rebuild `sharp` for my architecture:
 
     ```
     npm rebuild sharp --verbose
    ```
   
 4. **Verify the rebuild** – After rebuilding, I checked the folder again:
 
     ```
     ls -l build/Release
    ```
    This time, the correct binary (`sharp-darwin-arm64v8.node`) was there.

5. **Retry Ignite CLI** – With the rebuild done, I retried the Ignite CLI command, and it worked:

    ```
    npx ignite-cli@latest new <project-name>
    ```
    
Hopefully, this helps anyone else running into the same issue! It looks like the problem is caused by the way npx caches dependencies, so forcing the rebuild directly in the cache fixed it for me.
